### PR TITLE
AND and OR logic filter support for entity types in 3.3

### PR DESCRIPTION
**Pull request by borkarsaish65**
## Release Notes

- Features
  - Implement per-entity filter semantics (OR/AND) and propagate filterType through the filtering flow.
  - Add getArrayFilterOperator(filterType, key) to choose PostgreSQL operator (&& for OR/overlap, @> for AND/contains-all).
  - Add getFilterType(entityType) to normalize entity metadata filterType.
  - Expose per-entity filterType on objects returned by convertEntitiesForFilter.

- Bug fixes
  - Fix handleStringType to accept and apply filterType (previously ignored the third parameter), ensuring OR/AND behavior is applied consistently.
  - Ensure array/string filter semantics are applied consistently in validateAndBuildFilters and helpers.

- Code quality / improvements
  - Centralize filterType normalization and remove duplicated normalization logic.
  - Thread per-entity filterType through validateAndBuildFilters, handleArrayType, and conversion helpers.
  - Harden array operator selection with sensible defaults when meta/filterType is missing or malformed.
  - Maintain use of parameterized queries and optional chaining where appropriate.

- Validation & API
  - Add validation for req.body.meta.filterType in entity-type create/update flows (uppercased and restricted to OR|AND).

- Tests & docs (recommended)
  - Add unit tests for OR/AND behavior on array and string fields and edge cases (missing/malformed meta).
  - Add JSDoc for validateAndBuildFilters, getFilterType, getArrayFilterOperator, and related helpers to document mapping between filterType and PostgreSQL operators.

- Security (critical)
  - package.json dev script exposes Node debugger on 0.0.0.0:9229 — bind to localhost (127.0.0.1) or remove the explicit 0.0.0.0 binding immediately.

## Contributor Statistics

| Author | Lines Added | Lines Removed |
|--------|------------:|-------------:|
| *Unavailable* | *Could not retrieve per-author stats: branch feature/3.3stories not found locally* | *Could not retrieve per-author stats* |

### CODE REVIEW

**Comment by borkarsaish65** [comment]
@coderabbitai do code review , verify if the changes adhere to best practices and fail safe.

**Comment by borkarsaish65** [comment]
@coderabbitai review the PR freshly , goal is to adhere to best coding practices and fail safe code

File: src/generics/utils.js [hunk 1]
```diff
@@ -724,13 +732,15 @@ function convertEntitiesForFilter(entityTypes) {
 			})
 		}
 
+		const filterType = entityType?.meta?.filterType?.trim()?.toUpperCase() === 'OR' ? 'OR' : 'AND'
```
[the remark this answers is not in the document]
Comment (borkarsaish65) [reply src/generics/utils.js]: @coderabbitai I have addressed all the comments and made the necessary changes, review my PR and let me know if its approved frm your side

[the remark this answers is not in the document]
Comment (borkarsaish65) [reply src/generics/utils.js]: @coderabbitai for string type the filter will always be 'OR' , we dont need any change there.
Status cant be both 'Active' and 'inactive' right hence the logic

[the remark this answers is not in the document]
Comment (borkarsaish65) [reply src/generics/utils.js]: @coderabbitai is my PR approved from your side now ?

[the remark this answers is not in the document]
Comment (borkarsaish65) [reply src/generics/utils.js]: @coderabbitai suggest JSDoc for convertEntitiesForFilter and also provide a comment for handleStringType function

[the remark this answers is not in the document]
Comment (borkarsaish65) [reply src/generics/utils.js]: @coderabbitai made changes , give final approval

[the remark this answers is not in the document]
Comment (borkarsaish65) [reply src/generics/utils.js]: @coderabbitai I will ignore the final change as this is too big and I already added a comment explain the variable

**Comment by nevil-mathew** [comment]
As discussed earlier on the call, please make the necessary changes (validation, API documentation, etc.).

**Comment by borkarsaish65** [comment]
@coderabbitai full review

**Comment by borkarsaish65** [comment]
@coderabbitai full review and let me know if approved from your side.
